lib/galaxy/metadata/__init__.py

```diff
@@ -217,20 +287,17 @@ def __get_filename_override():
             # return command required to build
             fd, fp = tempfile.mkstemp(suffix='.py', dir=tmp_dir, prefix="set_metadata_")
             metadata_script_file = abspath(fp)
-            os.fdopen(fd, 'w').write('from galaxy_ext.metadata.set_metadata import set_metadata; set_metadata()')
+            os.fdopen(fd, 'w').write(SET_METADATA_SCRIPT)
```

Comment:
I think we'll want to close the descriptor here.

Reply:
Added to tracking branch (https://github.com/galaxyproject/galaxy/pull/7058/commits/c4da5abbe3fbce7eb058cc3ba9aeb415899b3a35).